vm/ubpf_vm.c

```diff
@@ -425,7 +425,7 @@ ubpf_exec_ex(
             goto cleanup;
         }
 
-        if (pc == 0 || vm->int_funcs[pc]) {
+        if ((pc == 0 || vm->int_funcs[pc]) && stack_frame_index < UBPF_MAX_CALL_DEPTH) {
```

Comment:
Interesting. So, what I am seeing is that `stack_frame_index` overflows and causes an out-of-bounds access to `stack_frames`? If that is the case, then this appears to solve the issue! 

A subsequent bpf2bpf call cannot happen because there is a guard with the very same condition? 

Again, if that is the case, then I am 100% on board with this change.

Reply:
Yes, exactly. The inner most call computes the stack usage, but it then writes outside the bounds. The next inner call would be blocked, so the value isn't actually neede.

Reply:
As a `TODO` we could _somehow_ add an optimization if there are leaf functions. Thank you for the response! I am approving!!